dll/win32/shell32/shlexec.cpp

```diff
@@ -2124,6 +2124,39 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
             *end = L'\0';
             lpFile = wfileName;
         }
+        /* We have to test sei instead of sei_tmp because sei_tmp had its
+         * input fMask modifed above in SHELL_translate_idlist.
+         * This code is needed to handle the case where we only have an
+         * lpIDList with multiple CLSID/PIDL's (not 'My Computer' only) */
+        else if ((sei->fMask & SEE_MASK_IDLIST) == SEE_MASK_IDLIST)
+        {
+            WCHAR buffer[MAX_PATH], xlpFile[MAX_PATH];
+            LPWSTR space, s;
+
+            LPWSTR beg = wszApplicationName;
+            for(s = beg; (space = const_cast<LPWSTR>(strchrW(s, L' '))); s = space + 1)
+            {
+                int idx = space - sei_tmp.lpFile;
+                memcpy(buffer, sei_tmp.lpFile, idx * sizeof(WCHAR));
+                buffer[idx] = '\0';
+
+                if (SearchPathW(*sei_tmp.lpDirectory ? sei_tmp.lpDirectory : NULL,
+                    buffer, L".exe", _countof(xlpFile), xlpFile, NULL))
```

Comment:
Should this only be `.exe` or also `.bat`, `.cmd` etc?

Reply:
I am not sure about this. My intent was just to fix the issue of https://jira.reactos.org/browse/CORE-18967 which was caused by a previous deletion of parts of this section of code in https://github.com/reactos/reactos/commit/0f16d44b660266130f5c0807c01d01cfaa6e685d. Do you need more than this? Thanks.